src/iologindata.h

```diff
@@ -58,6 +58,7 @@ class IOLoginData
 		static void removeVIPEntry(uint32_t accountId, uint32_t guid);
 
 		static void updatePremiumTime(uint32_t accountId, time_t endTime);
+		static void getUnjustifiedDates(std::string name, std::vector<time_t>& dateList, time_t offsetTime);
```

Comment:
This function should return the container instead.

```suggestion
		static std::vector<time_t> getUnjustifiedDates(std::string name, time_t offsetTime);
```

Reply:
@ranisalt Why it should? I'm not asking to deny your suggestion, but as I'm not that experienced in C, I wanted to know difference between returning vector and modyfing declared somewhere else.

Reply:
> but as I'm not that experienced in C,

Specifically because this is not C. In C, you would pass a pointer to the output variable and return the result (e.g. 0 if success, 1 if error), but in C++ we have exceptions, so a function should return its output and throw when an error occurs.

Besides that, there's a myriad of optimizations that can be performed by the compiler in return values that can't be applied with function parameters, because when you return a value you have several constraints (e.g. you don't know if the parameter vector came in empty or with some values, if it was allocated on the stack or heap...).

Overall, it's a better function signature. The function is called "get me something" but it does not return you anything just feels weird in this context :P

Reply:
By the way, you probably want to take in `const std::string& name` instead of `std::string name` to avoid a copy.